src/core/thread/neighbor_table.hpp

```diff
@@ -177,6 +177,17 @@ class NeighborTable : public InstanceLocator, private NonCopyable
     Neighbor *FindNeighbor(const Ip6::Address   &aIp6Address,
                            Neighbor::StateFilter aFilter = Neighbor::kInStateValidOrRestoring);
 
+    /**
+     * Searches in the neighbor table to find a `Neighbor` for which a one-way link is maintained (as in the
+     * case of an FTD child with neighbor routers).
```

Comment:
This is technically searching the router table, right?

Reply:
Yes. It has a guard to ensure device `IsChild()` and then `neighbor = Get<RouterTable>().FindNeighbor(aMacAddress);`